app/models/pydantic/datamart.py

```diff
@@ -1,12 +1,68 @@
 from enum import Enum
-from typing import Dict, Optional
+from typing import Dict, Literal, Optional, Union
 from uuid import UUID
+from abc import ABC, abstractmethod
 
-from pydantic import Field
+from pydantic import Field, root_validator, validator
 
 from app.models.pydantic.responses import Response
 
 from .base import StrictBaseModel
+from ...crud.geostore import get_gadm_geostore_id
+
+
+class AreaOfInterest(StrictBaseModel, ABC):
+    @abstractmethod
+    async def get_geostore_id(self) -> UUID:
+        """Return the unique identifier for the area of interest."""
+        pass
+
+
+class GeostoreAreaOfInterest(AreaOfInterest):
```

Comment:
One last comment:   I feel `CustomAreaOfInterest` or `UserAreaOfInterest` may be more fitting name for the public facing doc

Reply:
I personally feel that those are too vague. But I'd like to hear what other folks think.
CC @jterry64 @dmannarino

Reply:
We're going to keep the current naming for now. Will revisit in the future.